src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntTests.cs

```diff
@@ -40,10 +42,232 @@ public void LanguageVersion()
+    void F2(System.UIntPtr x, nuint y);
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+
+            var method = comp.GetMember<MethodSymbol>("I.F1");
+            Assert.Equal("void I.F1(System.IntPtr x, nint y)", method.ToDisplayString(FormatWithSpecialTypes));
+            VerifyTypes((NamedTypeSymbol)method.Parameters[0].Type, (NamedTypeSymbol)method.Parameters[1].Type, signed: true);
+
+            method = comp.GetMember<MethodSymbol>("I.F2");
+            Assert.Equal("void I.F2(System.UIntPtr x, nuint y)", method.ToDisplayString(FormatWithSpecialTypes));
+            VerifyTypes((NamedTypeSymbol)method.Parameters[0].Type, (NamedTypeSymbol)method.Parameters[1].Type, signed: false);
+        }
+
+        /// <summary>
+        /// System.IntPtr and System.UIntPtr definitions from source.
+        /// </summary>
+        [Fact]
+        public void TypeDefinitions_FromSource()
+        {
+            var sourceA =
+@"namespace System
+{
+    public class Object { }
+    public abstract class ValueType { }
+    public struct Void { }
+    public struct IntPtr { }
+    public struct UIntPtr { }
+}";
+            var sourceB =
+@"interface I
+{
+    void F1(System.IntPtr x, nint y);
+    void F2(System.UIntPtr x, nuint y);
+}";
+            var comp = CreateEmptyCompilation(new[] { sourceA, sourceB }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+            verify(comp);
+
+            comp = CreateEmptyCompilation(sourceA);
+            comp.VerifyDiagnostics();
+            var ref1 = comp.ToMetadataReference();
+            var ref2 = comp.EmitToImageReference();
+
+            comp = CreateEmptyCompilation(sourceB, references: new[] { ref1 }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+            verify(comp);
+
+            comp = CreateEmptyCompilation(sourceB, references: new[] { ref2 }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+            verify(comp);
+
+            static void verify(CSharpCompilation comp)
+            {
+                var method = comp.GetMember<MethodSymbol>("I.F1");
+                Assert.Equal("void I.F1(System.IntPtr x, nint y)", method.ToDisplayString(FormatWithSpecialTypes));
+                VerifyTypes((NamedTypeSymbol)method.Parameters[0].Type, (NamedTypeSymbol)method.Parameters[1].Type, signed: true);
+
+                method = comp.GetMember<MethodSymbol>("I.F2");
+                Assert.Equal("void I.F2(System.UIntPtr x, nuint y)", method.ToDisplayString(FormatWithSpecialTypes));
+                VerifyTypes((NamedTypeSymbol)method.Parameters[0].Type, (NamedTypeSymbol)method.Parameters[1].Type, signed: false);
+            }
+        }
+
+        private static void VerifyTypes(INamedTypeSymbol underlyingType, INamedTypeSymbol nativeIntegerType, bool signed)
```

Comment:
nit: I assume this is not a local function because you intend to re-use elsewhere in the future?

Reply:
It was used elsewhere previously. Another reason for keeping as a member of the type rather than a local function is consistency with the other overload.

---
In reply to: [375002430](https://github.com/dotnet/roslyn/pull/40994#discussion_r375002430) [](ancestors = 375002430)